test/benchmark/BenchmarkIngest_test.go

```diff
@@ -0,0 +1,192 @@
+package benchmark
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/v3io/v3io-tsdb/config"
+	"github.com/v3io/v3io-tsdb/pkg/tsdb"
+	"github.com/v3io/v3io-tsdb/pkg/utils"
+	"github.com/v3io/v3io-tsdb/test/benchmark/common"
+	"io/ioutil"
+	"log"
+	"os"
+	"testing"
+	"time"
+)
+
+func BenchmarkIngest(b *testing.B) {
+	log.SetFlags(0)
+	log.SetOutput(ioutil.Discard)
+	testStartTimeNano := time.Now().UnixNano()
+
+	var count = 0 // count real number of samples to compare with query result
+
+	testConfig, err := common.LoadBenchmarkIngestConfigFromData()
+	if err != nil {
+		panic(err)
+	}
+
+	v3ioConfigFile := os.Getenv(common.TsdbV3ioConfig)
+	v3ioConfig, err := config.LoadConfig(v3ioConfigFile)
+	if err != nil {
+		panic(errors.Wrap(err, fmt.Sprintf("Failed to load config from file %s", v3ioConfigFile)))
+	}
+
+	adapter, err := tsdb.NewV3ioAdapter(v3ioConfig, nil, nil)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	appender, err := adapter.Appender()
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	// run the runTest function b.N times
+	relativeTimeOffsetMs, err := utils.Str2duration(testConfig.StartTimeOffset)
+	if err != nil {
+		b.Fatal("Unable to resolve start time. Check configuration.")
+	}
+	testStartTimeMs := testStartTimeNano/int64(time.Millisecond) - relativeTimeOffsetMs
+	timeStampsCount := (testStartTimeNano/int64(time.Millisecond) - testStartTimeMs) / int64(testConfig.SampleStepSize)
+	timeStamps := make([]int64, timeStampsCount)
+
+	for i := range timeStamps {
+		timeStamps[i] = testStartTimeMs + int64(i*testConfig.SampleStepSize)
+	}
+
+	sampleTemplates := common.MakeSampleTemplates(
+		common.MakeSamplesModel(
+			testConfig.NamesCount,
+			testConfig.NamesDiversity,
+			testConfig.LabelsCount,
+			testConfig.LabelsDiversity,
+			testConfig.LabelValuesCount,
+			testConfig.LabelsValueDiversity))
+
+	samplesCount := len(sampleTemplates)
+	refsMap := make(map[uint64]bool, samplesCount)
+	refsArray := make([]uint64, samplesCount)
+
+	for i := 0; i < b.N; i++ {
+		rowsAdded := runTest(i, appender, b, timeStamps, sampleTemplates, testConfig.FlushFrequency, refsMap, refsArray,
+			testConfig.AppendOneByOne, testConfig.Verbose)
+		if rowsAdded == 0 {
+			break // stop the test (target has been achieved)
+		}
+		count += rowsAdded
+	}
+
+	b.Logf("\nTest complete. Count: %d\n", count)
+}
+
+func runTest(
+	index int,
+	appender tsdb.Appender,
+	testCtx *testing.B,
+	timeStamps []int64,
+	sampleTemplates []string,
+	flushFrequency int,
+	refsMap map[uint64]bool,
```

Comment:
Doesn't look like this needs to be a map. We always save `true` in the value and never use it...